### ISSUE TITLE
Resistance FOM Audio is "behind me"

### ISSUE THREAD

**Issue by chyrio** [RPCS3/rpcs3#10464]
Just got this game working and maybe its my headset. But the Audio "location" is like behind me. Music works, some sound effects work but for example when someone is talking to me I cant hear them until I turn my visual to the left so they are like in my left ear. Reload noises, grenade sounds, and melee sounds are the same story. I tried changing the audio from Xaudio2 to OpenAL and like 5.1 to stereo to 7.1.. No change.

**Comment by MSuih** [comment]
Did you try enabling downmix to stereo? What sort of headset do you have, does it have physical or virtual surround mode?

**Comment by chyrio** [comment]
> 
> 
> Did you try enabling downmix to stereo? What sort of headset do you have, does it have physical or virtual surround mode?

I did. I tried every setting I could change for audio.. no difference. The headset is a 5.1 channel Steelseries which does support virtual audio. But it seems to do the same thing with the local laptop speakers and my bluetooth speaker.

**Comment by chyrio** [comment]
Its almost like if you had a 5.1 channel stereo setup and was missing the "center" speaker.

**Comment by SolomonGrundyThanksForTheTurkey** [comment]
I've had this same issue, but closing and restarting the game usually fixes it, and every subsequent start up (unless you close RPCS3 itself).

**Comment by Megamouse** [comment]
There has been a fix to surround sound.
Maybe it was related.
Please re-test

**Comment by Lagahan** [comment]
Still happening, recorded a video;
The audio on the video is 5.1 downmixed to stereo via my soundcard - should be able to hear all channels, also shouldn't affect stereo output from RPCS3, it treats it as normal (just sounds a bit tinny from the spatial effect).

Downmix to stereo in RPCS3 settings sometimes misses the center, front and rear channels, only playing sides.
Downmix to 5.1 usually works but the FMVs always have corrupted audio. 
Happens in both cubeb and xaudio2.
Headphone warning for the FMV on 2nd boot of the game, its loud.
[https://www.youtube.com/watch?v=HC9Jm5zQopc](url)

Killzone 2 also misses channels sometimes in both 5.1 and stereo but has no issues with FMV audio in either. Couldn't replicate it tonight but it happened the other day testing the new optimization PRs.

**Comment by Lagahan** [comment]
Issue cropped up this morning on Killzone 2 when I was about to check something else:
[https://www.youtube.com/watch?v=XvWAKCQgXaM](url)

This time with downmix to 5.1. Also intermittent.

**Comment by Lagahan** [comment]
I'll give that a test after work, will need to run the games a few times because its intermittent.

**Comment by Lagahan** [comment]
Ok so, clearing PPU and SPU cache directly after taking that video of Killzone 2 fixed the missing channels.
Tried booting the game several times (while keeping RPCS3 open) after clearing the caches and it worked every time.
However when I restarted windows and tried again the same audio channels were missing each time I booted the game.

As for Resistance I couldn't replicate the missing channels when downmixing to stereo after clearing PPU and SPU caches, even after rebooting the machine a couple times.
Audio in FMVs is still very corrupted when downmixing to 5.1 though, and there's a good chance the missing channels wasn't really fixed by that given how intermittent the issue is.

**Comment by marcoluc97** [comment]
Megamouse can #12112 help them here? Even just more logging

**Comment by Megamouse** [comment]
Maybe

**Comment by Linear524** [comment]
An interesting audio bug sitting here a long time in The Last of Us - first escaping scene where Sarah riding in the car with Joel and Tommy, if you turn around 180 degrees on the backseat = every speech between Joel and Tommy is working loud and clear... But if you just sitting and watching the road as intended - almost every word between Joel and Tommy is silent, like front channels being muted for their speech...

**Comment by Megamouse** [comment]
Please try again

**Comment by Lagahan** [comment]
Just a heads up, I was having an issue with that PR (and after it merged) where saving custom config without booting a game first would crash the emulator, but when I deleted the main config.yml while testing this it fixed it.

Resistance is still missing channels in the new Surround 5.1 output on first boot (can't hear menu selection sounds), but when the game reboots and launches the main executable when loading into a level sound is fine. Broken audio in cutscenes in 5.1 is also fixed. 
Missing channels / sounds in Stereo seems to be fixed outright, but now the cutscenes sound like they have fake spatial surround applied. 
This could be original behavior though as it has 5.1ch cutscene AVIs, you can check the codec info on them in the game files. I'll have to hook up my PS3 to test it.

Killzone 2 still has missing center, LR and RR channels in Surround 5.1 on first boot sometimes - its inconsistent. Rebooting the game usually fixes it. FWIW in both games it was always those channels missing.
Stereo output seems reliably fine in this game as well though, with no cutscene weirdness.

**Comment by Linear524** [comment]
Killzone 2 and 3 works great, no issues.
TLoU - same speech fading issue in the car... But looks like it is SPU issues, or sound position is tied to actual framerate...
Anyway - good work with sound backend ! :)

**Comment by marcoluc97** [comment]
After #12164 any changes of behaviour?

**Comment by Lagahan** [comment]
No change, still missing channels on Killzone 2 and Resistance occasionally in `Surround 5.1`. I agree with Linear on the possible SPU source for this, I noticed this started way back when SPU LLVM came in. Could be a race issue or some other esoteric timing issue. Stereo still works fine though.

**Comment by Megamouse** [comment]
Please try with #12186

**Comment by Lagahan** [comment]
> Please try with #12186

Still inconsistent.
When I wasn't screen recording with OBS it didn't work first launch of the game on the new PR, then did work second launch of the game without restarting RPCS3.

In the video I restarted the emulator a few times; first 2 times it worked (you can hear the menu click on center channel when I spam X to skip the intro), so I kept restarting the emulator till it stopped working. 
Then I restarted it again til it did work (can hear the button spam in the intro again). 
[https://youtu.be/Q2wlWccl78o](url)

No idea whats going on here.

**Comment by marcoluc97** [comment]
With #12252 there is any difference? You can also try master as it is already been merged.

**Comment by Lagahan** [comment]
Still inconsistently missing center, rr and rl channels in 5.1. 

At this stage its probably something wrong with my machine if I'm the only one on 5.1 encountering this. At least stereo always works now.

**Comment by Megamouse** [comment]
Please try again

**Comment by Lagahan** [comment]
Still inconsistent in both games

**Comment by Megamouse** [comment]
please try again

**Comment by marcoluc97** [comment]
#12902 but already merged in master so just test lastest version

**Comment by Megamouse** [comment]
closing due to inactivity
